src/main/scala/outwatch/dom/helpers/DomUtils.scala

```diff
@@ -178,18 +174,45 @@ object DomUtils {
     }
   }
 
+
+  private final case class ChildrenNodes(
+    children: List[VNode_] = Nil,
+    hasStreams: Boolean = false
+  )
+  private def extractChildren(nodes: Seq[ChildVNode]): ChildrenNodes = nodes.foldRight(ChildrenNodes()) {
+    case (vn: VNode_, cn) => cn.copy(children = vn :: cn.children)
+    case (_: ChildStreamReceiver, cn) => cn.copy(hasStreams = true)
+    case (_: ChildrenStreamReceiver, cn) => cn.copy(hasStreams = true)
+  }
+
+  private[outwatch] def ensureVNodeKey(vn: VNode_): VNode_ = {
+    val withKey: VNode_ = vn match {
+      case vtree: VTree =>
+        val modifiers = vtree.modifiers
+        val hasKey = modifiers.exists(m => m.unsafeRunSync().isInstanceOf[Key])
```

Comment:
Is it right to `run` the `IO` here and then wrap the modifier itself again as an `IO`?

Reply:
`ensureVNodeKey` only gets called inside a snabbdom `patch`, so I think it's ok, it doesn't affect the referential transparency of the Outwatch API. 

Also `ensureVNodeKey` gets called from `extractChildrenAndDataObject` which gets called from `Vtree.asProxy` which already does `unsafeRunSync` on modifiers.

Note that it's not doing `.unsafeRunSync` and then wrapping inside `IO` again, it's just adding a `key` modifier to the existing modifiers.